compiler/ccgtypes.nim

```diff
@@ -1350,17 +1350,16 @@ proc genDisplay(t: PType, depth: int): Rope =
 
 proc genTypeInfoV2Impl(m: BModule; t, origType: PType, name: Rope; info: TLineInfo) =
   cgsym(m, "TNimTypeV2")
-  m.s[cfsStrData].addf("N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
+  m.s[cfsStrData].addf("extern N_LIB_PRIVATE TNimTypeV2 $1;$n", [name])
```

Comment:
C++ needs `extern`, see https://stackoverflow.com/questions/3714808/forward-declarations-for-variables